src/codegen_call.cpp

```
@@ -791,70 +792,70 @@ void CodeGen::storeMapHeaderFields(llvm::Value *headerPtr, llvm::Value *len,
+    cg.requireArgs(e, 1);
+    llvm::Value *x = cg.emitExpr(*e.args[0]);
+    if (x->getType() == cg.f64Ty_) {
+        auto fn = cg.getRuntimeFn("fabs", cg.f64Ty_, {cg.f64Ty_});
+        return cg.builder_.CreateCall(fn, {x}, "abs");
     }
     if (x->getType()->isIntegerTy(64)) {
-        llvm::Value *neg = builder_.CreateNeg(x, "neg");
-        llvm::Value *isNeg = builder_.CreateICmpSLT(x, llvm::ConstantInt::get(i64Ty_, 0), "is_neg");
-        return builder_.CreateSelect(isNeg, neg, x, "abs");
+        llvm::Value *neg = cg.builder_.CreateNeg(x, "neg");
+        llvm::Value *isNeg = cg.builder_.CreateICmpSLT(x, llvm::ConstantInt::get(cg.i64Ty_, 0), "is_neg");
```

Clang (14.0.6)

[warning] variable 'isNeg' is not initialized

(cppcoreguidelines-init-variables)